### PR TITLE
Update ESLint config and dependencies

### PR DESCRIPTION
**Pull request by Gudahtt**
All ESLint config and dependencies have been updated to the latest versions available. The `lint:fix` command has been run as well. All lint violations were fixed automatically.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,27 +1,13 @@
 module.exports = {
-
   root: true,
-
-  parserOptions: {
-    'ecmaVersion': 2017,
+  extends: ['@metamask/eslint-config', '@metamask/eslint-config-mocha'],
```
Comment (rekmarks) [inline .eslintrc.js:3]: We should only extend the Mocha config in the test file override.

**Review by rekmarks — CHANGES_REQUESTED** [review]
In addition to the inline comment, we should also add our default `.prettierrc.js`.

File: .eslintrc.js (hunk 1 above)
Comment (Gudahtt) [reply .eslintrc.js]: Whoops! Good catch. I had actually thought of this and checked the README for this config to see what was recommended, and the example there doesn't use an override. In that draft there was no override yet so I didn't bother. But now that it's here anyway, there's really no reason not to migrate it.

**Comment by Gudahtt** [comment]
Thanks! Both comments have been addressed

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**